gnucash/gnome-utils/gnc-main-window.c

```diff
@@ -1405,12 +1406,22 @@ gnc_main_window_quit(GncMainWindow *window)
     }
     if (do_shutdown)
     {
-        GList *w;
+        GList *w, *next;
 
-        for (w = active_windows; w; w = g_list_next (w))
+        for (w = active_windows; w; w = next)
         {
-            window = w->data;
-            window->window_quitting = TRUE; // set window_quitting on all windows
+            GncMainWindowPrivate *priv;
+            GncMainWindow *wind = w->data;
+
+            next = g_list_next (w);
```

Comment:
Why pull this out of the loop def?

Reply:
I thought there may be a chance that the 'next' pointer maybe removed as part of destroying the window as it removes the window from the active_windows list so I thought it would be safer. I can add a note to that effect if required.

Reply:
Right. Yeah, a comment would be good.